src/core/Geometry.js

```diff
@@ -799,21 +799,33 @@ Geometry.prototype = Object.assign( Object.create( EventDispatcher.prototype ),
 
 		for ( i = 0, il = uvs2.length; i < il; i ++ ) {
 
-			var uv = uvs2[ i ], uvCopy = [];
-
-			if ( uv === undefined ) {
-
-				continue;
-
-			}
-
-			for ( var j = 0, jl = uv.length; j < jl; j ++ ) {
-
-				uvCopy.push( uv[ j ].clone() );
-
-			}
-
-			uvs1.push( uvCopy );
+			var uv1 = uvs1[ i ], uv2 = uvs2[ i ];
+ 
+                	if ( uv1 === undefined || uv2 === undefined ) {
```

Comment:
But now it would only work with geometries with 2 uv sets, no?

Reply:
wait... no... uv1 and uv2 refer to self and the geometry to merge with. So if any of the two don't have a corresponding uvmap for "i" to merge, the merging is skipped. I tried to be consistent with the variable naming you used for the uvsets (uvs1, uvs2)

Reply:
> So if any of the two don't have a corresponding uvmap for "i" to merge, the merging is skipped.

So... if `uv2` is undefined then we no longer copy `uv1`. Do you think that's the correct behaviour?

Reply:
uv1 is never being copied because it's the uv data for of the geometry to merge into aka "this". If uv2 is undefined, which means that the geometry being merged with "this" does not have any uv data for the current index (three.js supports two uv data sets normally). So uv2 won't be appended to uv1. If uv1 is undefined there is no uv data for the current index in "this" to begin with so uv2 should not be appended.

uvs1[] and uv1 refer to uv data in "this"
uvs2[] and uv2 refer to uv data in the geometry passed in the merge function

I understand it looks a bit confusing but this is the naming that was already there so I build upon that.